signal-hook-registry/src/lib.rs

```diff
@@ -167,6 +213,38 @@ impl GlobalData {
     }
 }
 
+#[cfg(windows)]
+extern "C" fn handler(sig: c_int) {
+    if sig != SIGFPE {
+        // Windows CRT `signal` resets handler every time, unless for SIGFPE.
+        // Reregister the handler to retain maximal compatibility.
+        // Problems:
+        // - It's racy. But this is inevitably racy in Windows.
+        // - Interacts poorly with handlers outside signal-hook-registry.
+        unsafe {
+            libc::signal(sig, handler as sighandler_t);
+        }
+    }
+
+    let signals = GlobalData::get().all_signals.peek_signal_safe();
+
+    if let Some(ref slot) = signals.get(&sig) {
+        let fptr = slot.prev;
+        if fptr != 0 && fptr != SIG_DFL && fptr != SIG_IGN {
+            // FFI ‒ calling the original signal handler.
+            unsafe {
+                let action = mem::transmute::<usize, extern "C" fn(c_int)>(fptr);
```

Comment:
I wonder, if it is possible to convert in one direction with `as` (eg. the `handler as sighandler_t`), would it work the other way around too, instead of using transmute?

Reply:
That must not be the case, because `usize` allows 0 while `fn` doesn't allow 0 (instant UB, IIRC)

Reply:
Ah, I see. That makes sense.